src/test/java/org/terasology/logic/health/RegenTest.java

```diff
@@ -99,4 +106,32 @@ public void zeroRegenTest() {
 
         assertFalse(player.hasComponent(RegenComponent.class));
     }
+
+    @Test
+    public void regenTest() {
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 100;
+        healthComponent.maxHealth = 100;
+        healthComponent.waitBeforeRegen = 1;
+        healthComponent.regenRate = 1;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        player.send(new DoDamageEvent(5));
+        helper.runWhile(200, () -> true);
```

Comment:
Given a no-op, could this be Thread.sleep?

Reply:
Maybe? :thinking: I don't fully get how the game ticks work here... this is probably not needed, and just an artifact of me tinkering around ... :monocle_face: 

I'll do another round over these tests anyways, will see how much I can throw away then :sweat_smile: